devel/namespaces/index.md

```diff
@@ -0,0 +1,167 @@
+
+* `buildbot-bridge:`
+   The release engineering team's buildbot bridge (BBB) uses scopes with this prefix.
+
+* `project:<project>:`
+   Individual projects should use scopes with this prefix.
+
+* `signing:`
+   The release engineering team's signing system uses scopes with this prefix.
+
+## Roles
+
+Most roles are defined by some kind of automatic usage in a TaskCluster component.
+However, some are defined by convention.
+Both are listed here:
+
+* `client-id:<clientId>` -
+   Roles with this prefix give the scopes for client credentials.
+
+* `hook-id:<hookGroupId>/<hookId>` -
+   Roles of this form give the scopes used to create tasks on behalf of hooks.
+
+* `mozilla-group:<groupName>` -
+   Roles of this form represent the scopes available to members of the given Mozilla LDAP group via the login service.
+
+* `mozilla-user:<userName>` -
+   Roles of this form represent the scopes available to the given Mozilla LDAP user via the login service.
+
+* `repo:<host>/<path>:branch:<branch>`,
+* `repo:<host>/<path>:pull-request` -
+   Roles of this form represent scopes available to version-control pushes and pull requests.
+
+* `moz-tree:level:<level>`
+   Roles of this form include the basic scopes available to version-control trees at each of the three Mozilla source-code managament levels.
+   They are useful as shorthand to configure `repo:*` roles.
+   See [Mozilla Commit Access Policy](https://www.mozilla.org/en-US/about/governance/policies/commit/access-policy/) for information on levels.
+
+* `scheduler-id:<schedulerId>/<taskGroupId>` -
+   Roles of this form represent scopes available to schedulers with the given ID.
+
+* `worker-id:<workerGroup>/<workerId>` -
+   Roles of this form represent scopes available to workers with the given ID.
+
+* `worker-type:<provisionerId>/<workerType>` -
+   Roles of this form represent scopes available to workers of the given type.
+
+## Artifacts
+
+Artifacts are named objects attached to tasks, and available from the queue component.
+Artifact names are, by convention, slash-separated.
+
+* `public/` -
+   The queue allows access to any artifact that begins with `public/` without any kind of authentication.
+   Such public names are not further namespaced: tasks can create any public artifacts they like.
+
+* `private/` -
+   Artifact names with this prefix are considered non-public, but access to them is otherwise quite broadly allowed (e.g., to all Mozilla employees).
+   In general, users with narrower requirements than "not public" should select a different prefix and add it to this document.
+
+* `index/` -
+   Artifact names with this prefix follow the index namespace, replacing the `.` separators with `/`.
+   For example, a task that indexes artifacts under `gaia.npm_cache.node-v4-2.linux-x64` would name its artifacts with the prefix `index/gaia/npm_cache/node-v4_2/linux-x64`.
+   Scopes for creating and reading such artifacts should follow the ownership hierarchy of the index routes.
```

Comment:
We would have to keep the scopes in sync... and many of these scopes are automatically generated in decision tasks etc...

Let's add if it there is a use-case... but I don't see one... Generally I don't want people to use non-public artifacts unless strictly required... We can cache private artifacts, but at the moment we don't and doing so would be a lot of work...